test/jwt-decode-middleware-test.js

```diff
@@ -1,46 +1,43 @@
-import { expect } from 'chai'
+import { assert } from 'chai'
 import jwt from 'jsonwebtoken'
 import jwtDecodeMiddleware from '../src/jwt-decode-middleware'
 
 describe('JWTDecodeMiddleware', () => {
-  let middleware = jwtDecodeMiddleware()
-  let req
-  let token
+  let middleware = jwtDecodeMiddleware({
+    algorithm: 'HS256',
+    secret: 'some secret'
+  })
 
-  before((done) => {
-    const options = {
+  it('decodes valid JWT token', (done) => {
+    const token = jwt.sign({ data: 'Hello world' }, 'some secret', {
       algorithm: 'HS256'
-    }
-    jwt.sign('Hello world', 'secret', options,
-      (err, result) => {
-        token = result
-        done(err)
-      })
-  })
+    })
+    let req = { headers: { authorization: `bearer ${token}` } }
 
-  it('decodes JWT token', (done) => {
-    req = {
-      headers: {
-        authorization: `bearer ${token}`
+    middleware(
+      req,
+      {},
+      () => {
+        assert.deepOwnInclude(req.jwt, { data: 'Hello world' })
+        done()
       }
-    }
-    const next = () => {
-      expect(req.jwt).to.equal('Hello world')
-      done()
-    }
-    middleware(req, {}, next)
+    )
   })
 
-  it('decodes JWT token without secret', (done) => {
```

Comment:
Why is this test removed?

Reply:
Because it should not trust an unsigned JWT token.

Reply:
Maybe a test for that behaviour then?

Reply:
Done.